### DISCUSSION TITLE
Provide Import  Excel Extension Using Pola.rs

### DISCUSSION THREAD

**Discussion by Deanozk** [duckdb/duckdb#12397] [Ideas]
1Now that Polar.rs has an native read_excel() function it seems it would be straightforward to provide an extension that uses Pola.rs as the API to bring the exel document into DuckDB.   It coud be done in a few lines of code and would be far more robust than what we have now.  Users should not have to install a geometric package have access to exel spreadsheets.  Let's just add to the exel extension the polars integration.  

Other options include: xlsx2csv  since DuckDB already has excelent csv support

Benefits

- A Robust Import Facility requiring little maintenace who's only point of failure is Pola.rs. 
- Users dont have to install separate unrelated packages to get access to excel spreadsheet support.
- DuckDB does not need to maintain pola.rs and integration with pola.rs is necessity anyway.
- The Key Library for doing this is Rust anyway. https://github.com/tafia/calamine  
- Pola.rs is guaranteed to support these formats anyway and the best library for it is in rust.  
- Even if DuckDB later has its own direct Calamine implementation this is still a step in that direction we can take now and use the same interface. 

Cons
   - You need to install the dependencies that pola.rs has for the library. It does not work without [all] or something extra. 
   - May need to wrap the Polar(s) request to include the right parameters.  See https://github.com/pola-rs/polars/issues/10380

**Comment by Deanozk** [comment]
``` 
import polars as pl

df = pl.read_excel(
    source="myspreadsheet.xlsx", schema_overrides={"date": pl.Datetime}
) ````

**Reply by pdet** [reply]
It should be possible to read the polars dataframe to achieve that, no?
https://duckdb.org/docs/guides/python/polars.html

**Reply by Deanozk** [reply]
Yes, it should be possible to read in the polars dataframe and  check that that read worked ok.

**Reply by Deanozk** [reply]
I do notice something about converting to polars Datetime and that would have to be looked into as well

**Reply by Deanozk** [reply]
https://docs.pola.rs/py-polars/html/reference/api/polars.read_excel.html

**Reply by Deanozk** [reply]
https://stackoverflow.com/questions/77893543/why-does-polars-pl-read-excel-make-sometimes-unknown-cause-error

**Reply by Deanozk** [reply]
It might be necessary to specify the calamine engine in doing this: https://github.com/pola-rs/polars/issues/10380

**Reply by Deanozk** [reply]
> It should be possible to read the polars dataframe to achieve that, no? https://duckdb.org/docs/guides/python/polars.html

I think another possibility is xlsx2csv due to the excellent support DuckDB has for csv already

**Comment by Maxxen** [comment]
I don't think we want to add a dependency on polars. But you might be interested to know that [I've been working on a native excel/xml extension for DuckDB](https://github.com/Maxxen/duckdb-xml) as well that can stream larger-than-memory excel/xml files, although I don't have a timeline for when it will be ready.
